src/shared/tpm2-util.c

```diff
@@ -104,25 +104,20 @@ int dlopen_tpm2(void) {
                         DLSYM_ARG(Tss2_MU_TPM2B_PUBLIC_Unmarshal));
 }
 
-void tpm2_context_destroy(struct tpm2_context *c) {
-        assert(c);
-
-        if (c->esys_context)
-                sym_Esys_Finalize(&c->esys_context);
+struct tpm2_context *tpm2_context_free(struct tpm2_context *c) {
+        if (c) {
```

Comment:
we prefer early exit over deep nesting, hence maybe do:

```c
if (!c)
        return NULL;
```

so that you don't need to nest the whole function so deeply